src/java/picard/analysis/SinglePassSamProgram.java

```diff
@@ -138,8 +138,7 @@ public static void makeItSo(final File input,
     protected boolean usesNoRefReads() { return true; }
 
     /** Should be implemented by subclasses to do one-time initialization work. */
-    protected abstract void setup(final SAMFileHeader header, final File samFile);
-
+    protected abstract void setup(final SAMFileHeader header, final File samFile, final File referenceSequence);
```

Comment:
I don't think you really need to pass along referenceSequence here.  The only method that calls setup 'makeItSo', has just passes the global 'REFERENCE_SEQUENCE' directly to setup.

It would probably simplify the code if you just used REFERENCE_SEQUENCE itself within the setup method of CollectGcBiasMetrics